extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -225,11 +225,11 @@ private boolean hasFTAnnotations(IndexView index, AnnotationStore annotationStor
 
         // then check on the parent
         DotName parentClassName = info.superName();
-        ClassInfo parentClassInfo = index.getClassByName(parentClassName);
-        if (parentClassName.equals(DotNames.OBJECT) || parentClassInfo == null) {
-            return false; //no more parents
+        if (parentClassName == null || parentClassName.equals(DotNames.OBJECT)) {
+            return false;
         }
 
+        ClassInfo parentClassInfo = index.getClassByName(parentClassName);
```

Comment:
I think you should still test for a null return value here.

Reply:
Ok, we're never safe enough